### ISSUE TITLE
Barrier not starting up on macOS Big Sur 11.0.1

### ISSUE THREAD

**Issue by raffimohammed** [debauchee/barrier#958]
**Describe the bug**
Barrier not starting up after upgrading to macOS Big Sur 11.0.1 

**To Reproduce**
Start the Barrier application

Start the Barrier application

**Expected behavior**

**Screenshots**

Please do not link to image hosting sites, as these can be ephemeral. Instead, attach them to the issue. 

**Desktop (please complete the following information):**

 - OS: [macOS Big Sur]
- Barrier version [2.3.3]

**Additional context**

**Comment by DannyCork** [comment]
working fine on this Big Sur installation.

**Comment by kierandelaney** [comment]
I had a very similar issue - eventually after a restart I started getting the prompt to grant Accessibility rights - which were already ticked. Whatever I did, app launch triggered this request. I unticked and removed the app, started it, granted permission again and it finally opened. I am not sure what happened to the permissions, it was like Barrier couldn't see they were already set. 🤷‍♂️

Might be completely different, but I also had the same system log entries before the restart.

**Comment by raffimohammed** [comment]
Restart did not help. I uninstalled and installed the application which prompted me for Barrier to use Accessibility privileges (Settings --> Security & Privacy --> Privacy). I unticked and removed the older application; and gave access to the newer application. Then I was able to start the application. 

Here is the log after the successful launch of the application.

Nov 27 10:48:49 computer1 com.apple.xpc.launchd[1]: Coalition Cache Hit: app<application.barrier.64501363.64501490(501)> [5871]

**Comment by kierandelaney** [comment]
I upgraded a second system at the weekend to Big Sur and exactly the same thing happened. This time, I went straight to removing the permission and re-adding it to fix.

**Comment by jacob-labs** [comment]
> Restart did not help. I uninstalled and installed the application which prompted me for Barrier to use Accessibility privileges (Settings --> Security & Privacy --> Privacy). I unticked and removed the older application; and gave access to the newer application. Then I was able to start the application.
> 
> Here is the log after the successful launch of the application.
> 
> Nov 27 10:48:49 computer1 com.apple.xpc.launchd[1]: Coalition Cache Hit: app<application.barrier.64501363.64501490(501)> [5871]

Same issue here. The application starts, but the `Barrier is starting` thing in the bottom left never goes away

**Comment by jchien83** [comment]
On Big Sur 11.0.1 and Barrier v2.3.3 I am able to get the server running on macOS but when using the client it's stuck on Barrier is starting.

**Comment by raffimohammed** [comment]
@jchien83, can you delete the client application's privileges (Settings --> Security & Privacy --> Accessibility)? Don't just uncheck and check back the checkbox beside the Barrier application logo. Delete it by using the  "-" minus sign. When you restart the Barrier application it should prompt you to provide the application Accessibility privileges.

**Comment by jchien83** [comment]
> @jchien83, can you delete the client application's privileges (Settings --> Security & Privacy --> Accessibility)? Don't just uncheck and check back the checkbox beside the Barrier application logo. Delete it by using the "-" minus sign. When you restart the Barrier application it should prompt you to provide the application Accessibility privileges.

I got it working.  It helps if I look at the logs.  It was complaining about the Macbook name being rejected.

**Comment by JJFrampton** [comment]
> I had a very similar issue - eventually after a restart I started getting the prompt to grant Accessibility rights - which were already ticked. Whatever I did, app launch triggered this request. I unticked and removed the app, started it, granted permission again and it finally opened. I am not sure what happened to the permissions, it was like Barrier couldn't see they were already set. 
> 
> Might be completely different, but I also had the same system log entries before the restart.

This worked for me instantly, thanks.

I didn't have to remove the the app, just unchecked it, then relaunched the app and granted permissions again.

:+1:

**Comment by ericlowry** [comment]
Ok, the trick for me was to un-tick the permissions and actually remove Barrier from the "Allow the apps below to control your computer" list.  I clicked the minus button '-' at the bottom of the list to completely remove Barrier from the list altogether.  And then I clicked the add button '+' (right next to minus) and add Barrier back into the list.  Worked like a charm, no reboot required...

**Comment by emacstheviking** [comment]
Doesn't work for me, keeps saying 'Starting' but the client laptop won't connect, keeps on saying 'ERROR code 9'. I tried all the other remedies in this thread as well.

**Comment by raffimohammed** [comment]
> Doesn't work for me, keeps saying 'Starting' but the client laptop won't connect, keeps on saying 'ERROR code 9'. I tried all the other remedies in this thread as well.

Can you connect to server_ip:24800 from the client laptop?

**Comment by emacstheviking** [comment]
No, I used to server to verify IP then on client I unticked auto config, entered 192.168.0.2:24800 and it still eventually timed out with error 9.
I have the Firewall disabled on both the client and the server.

**Comment by raffimohammed** [comment]
> No, I used to server to verify IP then on client I unticked auto config, entered 192.168.0.2:24800 and it still eventually timed out with error 9.
> I have the Firewall disabled on both the client and the server.

Can you ping 192.168.0.2 from the client? ..and nc -vz 192.168.0.2 24800 from the client?
